### ISSUE TITLE
add support for private ipv4 address in subnet mappings of resource aws_lb

### ISSUE THREAD

**Issue by KadenLNelson** [terraform-providers/terraform-provider-aws#11403]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* aws_lb

### Potential Terraform Configuration

```hcl
resource "aws_lb" "test" {
  name               = "test-lb-tf"
  internal           = true
  load_balancer_type = "network"
  subnet_mapping {
    subnet_id = "subnet-08dece37dd0d4a0b3"
    private_ipv4_address = "10.0.1.15" # <===
  }
}
```

### References

* N/A

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/11044

**Comment by bflad** [comment]
Hi folks 👋  This functionality has been merged and will release with version 3.1.0 of the Terraform AWS Provider, next Thursday. Thanks to @KadenLNelson for the implementation. 👍
